pkg/sessionctx/variable/session.go

```diff
@@ -3878,3 +3879,17 @@ const (
 func (s *SessionVars) GetOptObjective() string {
 	return s.OptObjective
 }
+
+// UseLowResolutionTSO indicates whether low resolution tso could be used for execution.
+func (s *SessionVars) UseLowResolutionTSO() bool {
+	return !s.InRestrictedSQL && s.lowResolutionTSO
+}
+
+// PessimisticLockRequired indicates whether pessimistic locks are required for the following execution.
+func (s *SessionVars) PessimisticLockRequired() bool {
```

Comment:
The function name doesn't match its body without checking its usages. Is it better to rename it to something like `PessimisticLockEligibleForSelectLock`?

Reply:
Changed to `PessimisticLockEligible` to indicate whether the locks should not be ignored when `for update` clause is used.